cardano-node/src/Cardano/Node/Protocol/Types.hs

```diff
@@ -11,22 +11,28 @@ module Cardano.Node.Protocol.Types
   , SomeConsensusProtocol(..)
   ) where
 
-import           Cardano.Prelude
+import           Prelude
+import           Cardano.Prelude (Generic, NFData)
 
-import           Control.Monad.Fail (fail)
 import           Data.Aeson
 import           NoThunks.Class (NoThunks)
 
 import qualified Cardano.Api.Protocol.Types as Cardano
 
 import           Cardano.Node.Orphans ()
 import           Cardano.Node.Queries (HasKESInfo, HasKESMetricsData)
-import           Cardano.Tracing.Constraints (TraceConstraints)
+import           Cardano.Node.TraceConstraints (TraceConstraints)
+
 
 data Protocol = ByronProtocol
               | ShelleyProtocol
               | CardanoProtocol
-  deriving (Eq, Show, Generic)
+  deriving (Eq, Generic)
+
+instance Show Protocol where
+  show ByronProtocol = "Byron"
+  show ShelleyProtocol = "Shelley"
+  show CardanoProtocol = "Byron; Shelley"
```

Comment:
Out of curiosity, what's wrong with `"CardanoProtocol"`?

Reply:
I have not decided this, but kept it as is.

Reply:
This is historical -- it originates from the early days of the HFC:

https://github.com/input-output-hk/cardano-node/commit/52910ef5d6b32e5138f8f0b7338074eab34a2f48#diff-500fbe07caf3228c5a6da5afb0110a5ba5852c604b187bbca793d85a348e13a1R6